code/game/objects/items/devices/magnetic_lock.dm

```diff
@@ -11,46 +11,51 @@
 	icon = 'icons/obj/magnetic_locks/centcom.dmi'
 	icon_state = "inactive"
 	w_class = 3
-	req_access = list(103)
+	req_access = list(access_cent_specops)
 	health = 90
 
 	var/department = "CENTCOM"
 	var/status = 0
+	var/locked = 0
+	var/hacked = 0
```

Comment:
You are correct, I was a bit too quick reading through the code.
If you want to work with bitflags, then you could define two additional states for locked and hacked and then use bitwise operations. to check if the maglock is in a specific state